crates/swc_ecma_transforms_base/src/hygiene/usage_analyzer.rs

```diff
@@ -313,36 +344,73 @@ impl UsageAnalyzer<'_> {
 
         let id = self.get_renamed_id(id);
 
-        let need_rename = {
+        let renames = {
             let b = self.cur.data.decls.borrow();
             let ctxts_of_decls = b.get(&id.0);
 
             if let Some(ctxts_of_decls) = ctxts_of_decls {
-                let cur_scope_conflict = if ctxts_of_decls.iter().any(|(_, ctxt)| ctxt == &id.1) {
-                    ctxts_of_decls.len() > 1
-                } else {
+                let conflicts: Vec<DeclInfo> = ctxts_of_decls
+                    .iter()
+                    .filter(
+                        |DeclInfo {
+                             ctxt,
+                             block_scoped: bs,
+                             ..
+                         }| ctxt == &id.1 && (block_scoped || *bs),
```

Comment:
@kdy1 
Do you mean that if we detect some specific SyntaxContext, we should treat it as `private_ident` as well?

Reply:
Yes, actually `SyntaxContext` is the trick behind `private_ident!`. 

`private_ident!` works by creating a unique `SyntaxContext` (by `SyntaxContext::empty(Mark::fresh(Mark::root()))`), and creating `Ident` witth provided symbol and the `SyntaxContext` we made.

Actual renaming is done by the `hygiene` pass.